File: TennisInObjectiveC/TennisInObjectiveC/Tennis Files/MatchScore.m
//
//  MatchScore.m
//  TennisInObjectiveC
//
//  Created by Stephen Kyles on 1/31/14.
//  Copyright (c) 2014 Stephen Kyles. All rights reserved.
//

#import "MatchScore.h"

@implementation MatchScore

-(instancetype) initWithFirstPlayer: (Player *) p1 secondPlayer: (Player *) p2
{
    if( (self = [super initWithFirstPlayer:p1 secondPlayer:p2   ]) == nil)
        return nil;
    return self;
}

-(BOOL) haveAWinner
{
    return self.player1Score == 3 || self.player2Score == 3;
}

-(void)addScore:(Score *)score
{
    self.scores[setNumber] = (SetScore *)score;
    [score getWinner] == self.player1 ? self.player1Score++ : self.player2Score++;
    [self.scores addObject:[score getWinner]];
    setNumber++;
}

-(NSString *) description
{
    NSLog(@"Set No.    Player A     Player B\n");
    for( int i = 0; i < setNumber; i++ ) {
        NSLog(@"   %d          %d            %d", i+1, self.player1Score, self.player2Score);
    }
    return self.player1Score > self.player2Score ?
    [NSString stringWithFormat:@"\n\nplayer 1 wins %d to %d\n\n", self.player1Score, self.player2Score] :
    [NSString stringWithFormat:@"\n\nplayer 2 wins %d to %d\n\n", self.player2Score, self.player1Score];
}

@end

File: CS 470/Project 1/Tennis/MatchScore.cpp
#include "MatchScore.hpp"
#include "iostream"
#include "iomanip"
using namespace std;

MatchScore::MatchScore( Player *p1, Player *p2 ): Score( p1, p2 ), setNumber(0) {}

bool MatchScore::haveAWinner()  {
    return player1Score() == 3 || player2Score() == 3;
    
}

void MatchScore::addScore( Score *score ) {
    scores[setNumber] = reinterpret_cast<SetScore *>(score);
    setNumber++;
    score -> getWinner() == player1() ? p1Score++ : p2Score++;
}

void MatchScore::print() {
    cout << "   Set No.    Player A          Player B\n";
    for( int i = 0; i < setNumber; i++ ) {
        cout << setw(7) << i+1;
        scores[i]->print();
        cout << endl;
    }
    if ( player1Score() > player2Score() ) {
        cout << "\nPlayer A wins the match " << player1Score() << " sets to "
        << player2Score() << endl;
        cout << endl;
    }
    else{
        cout << "\nPlayer B wins the match " << player2Score() << " sets to "
        << player1Score() << endl;
        cout << endl;
    }
}

MatchScore::~MatchScore() {
    for( int i = 0; i < setNumber; i++ )
        delete scores[i];
}
